### DISCUSSION TITLE
install with composer1 and composer2 both your requirements could not be solved

### DISCUSSION THREAD

**Discussion by SolsWebdesign** [kimai/kimai#3750] [On-Premise: Installation & Updates]
Hello I have used Kimai for a bit and am setting up a new version of it using the install guide on https://www.kimai.org/documentation/installation.html and I have php8.1 and composer 2 and found out that it could not be installed using composer 2. So I installed composer 1 (not really happy about it but so be it) and ... found out that does not work either with simply a different set of errors:
most of the errors complain about requires composer-runtime-api ^2 which I looked up and there they want Composer 2...

So composer 2 does not work (also see this issue https://github.com/kimai/kimai/issues/1731 about composer 2 errors like this one:
Problem 3
- symfony/flex is locked to version v1.6.2 and an update of this package was not requested.
- symfony/flex v1.6.2 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
) 

and composer 1 has its own set of errors like these : 
Problem 1
    - Installation request for doctrine/dbal 3.5.2 -> satisfiable by doctrine/dbal[3.5.2].
    - doctrine/dbal 3.5.2 requires composer-runtime-api ^2 -> no matching package found.
  Problem 2
    - Installation request for doctrine/migrations 3.5.2 -> satisfiable by doctrine/migrations[3.5.2].
    - doctrine/migrations 3.5.2 requires composer-runtime-api ^2 -> no matching package found.
  Problem 3
    - Installation request for doctrine/orm 2.14.0 -> satisfiable by doctrine/orm[2.14.0].
    - doctrine/orm 2.14.0 requires composer-runtime-api ^2 -> no matching package found.

Is there a very specific composer version that I need to use?

**Comment by kevinpapst** [comment]
Kimai works with Composer 2 - haven't used Composer 1 for quite a while, so I cannot anything about it. But I'd say (and the errors do too) it is not supported anymore, due to the dependencies. You are linking an issue that is more than 2 years old and drawing wrong conclusions from that.

Aside from that: I have no clue what you are doing. Kimai is installed daily dozens of times. Please re-read the docs and try it again with Composer 2.  Then please post the entire bash output and not just some random bits.

**Comment by SolsWebdesign** [comment]
Thank you Kevin for your fast answer. So I played around with composer (I had 2.0.7 installed and it is old) and now have installed composer 2.5-dev (and still on php 8.1) and it works a lot better but at the end I get a 
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 255

Maybe too new maybe something else, I'll dig around. Found this  https://github.com/kimai/kimai/issues/3456 so I could have a typo somewhere but I tried variations. I'll try again with a composer that is not dev (because I used 2.5-dev now)...
thank you for your help and your fast answer, very appreciated

**Reply by kevinpapst** [reply]
Clearing cache is no problem of composer.
Just do a `rm -r var/cache/*` that will erase the cache entirely as well.
Could either be caused by permissions or memory limit (which you can set when executing composer via php environment setting).

**Reply by SolsWebdesign** [reply]
COMPOSER_MEMORY_LIMIT=-1 composer install --optimize-autoloader -n

worked like a charm! Thank you very much :-)
